spec/features/building_map_spec.rb

```diff
@@ -45,33 +45,43 @@
       expect(page).to have_selector("div.building-icon", minimum: 13)
     end
 
-    it "shows the pin of a target point", js: true do
-      visit building_map_path(target: "52.393913,13.133082")
-      expect(page).to have_css('.target-pin')
+    # it "shows the pin of a target point", js: true do
+    #   visit building_map_path(target: "52.393913,13.133082") # TODO: This doesn't work anymore
+    #   expect(page).to have_css('.target-pin')
+    # end
+
+    it "shows no route, if it's not requested", js: true do
+      visit building_map_path
+      expect(page).not_to have_css(".routing-path")
+      expect(page).not_to have_css(".time-icon")
     end
 
     context "with route" do
-      it "shows a calculated route", js: true do
-        visit building_map_path(start: "52.393913,13.133082", dest: "52.393861,13.129606")
-        expect(page).to have_css(".routing-path")
-      end
-
-      it "shows no route, if it's not requested", js: true do
+      before do
         visit building_map_path
-        expect(page).not_to have_css(".routing-path")
-        expect(page).not_to have_css(".time-icon")
+        find("#nav-link-navigation").click
+        fill_in 'start', with: 'Haus A'
+        fill_in 'dest', with: 'Haus L'
+        click_on 'Go'
+        sleep 4
```

Comment:
Is there a way to wait until the request is finished instead of waiting for a fixed amount?

Reply:
The sleep itself was just a desperate attempt to fix this but it doesn't seem to work. What's confusing me is, that the other test that calculates a route is working here (there are 3 tests with routes).
Unfortunately I don't know how to wait for the execution of a js method because it seems as javascript and ruby don't like each other that much.
I tried approaches from [this article](https://stackoverflow.com/questions/12555848/capybara-integration-testing-with-asynchronous-javascript) but they don't even run locally.

Reply:
I also tried to fix this by looking at the log from the js code (maybe something is wrong with that) but I can't get this output from the test. Tried some approaches there as well but they don't even work locally and the methods I found online are outdated.

Reply:
latest run [looks good](https://github.com/hpi-swt2/compass-portal-blue/runs/4766149985?check_suite_focus=true) 
Would you add some comments to the test to document the possibly flaky behaviour? So that if the test fails in the future, there are initial ideas as to why this might be the case